lib/ui/window.dart

```diff
@@ -286,6 +326,29 @@ class Locale {
     }
   }
 
+  /// The variants subtags for the locale.
+  ///
+  /// Each subtag is expected to be a valid Unicode Language Identifier variant
+  /// subtag that is listed in Unicode CLDR supplemental data:
+  /// http://unicode.org/cldr/latest/common/validity/variants.xml. Please see
+  /// constructor documentation.
+  Iterable<String> get variants {
+    if (_variants == null)
+      return const <String>[];
+    return _variants.split('-');
+  }
+
+  /// Hyphen-separated variants subtags for the locale.
+  ///
+  /// This can be null.
+  ///
+  /// This is expected to be a hyphen-separated sorted list of valid Unicode
+  /// Language Identifier variant subtags that are listed in Unicode CLDR
+  /// supplemental data:
+  /// http://unicode.org/cldr/latest/common/validity/variants.xml. Please see
+  /// constructor documentation.
+  final String _variants;
```

Comment:
Discussion point: is it reasonable enough to store the variants as a string and then do splitting on demand?

This requires the constructor to submit sorted subtags too, or the code to call a (to-be-written) canonicalize() or similar that will do the sorting while constructing a new Locale.

Reply:
I don't really understand why we would want to store these in this half-encoded form, especially if we're saying it has to be pre-sorted, because that implies that anyone creating a locale has to first decode their locale, split it up, sort it, then re-encode it, just so that we can subsequently re-split it. That seems... inefficient.